### PR TITLE
Fix/silence warnings related to c++98 compat

### PR DESCRIPTION
**Pull request by TroubleMakerBen**

### CODE REVIEW

**Comment by TroubleMakerBen** [comment]
Merge after #269

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Utils/MSErrorLogFormatter.m [hunk 1]
```diff
@@ -709,15 +711,15 @@ + (MSBinaryImageType)imageTypeForImagePath:(NSString *)imagePath processPath:(NS
     NSString *appBundleContentsPath = [standardizedImagePath substringToIndex:appRange.location + 5];
 
     if ([standardizedImagePath isEqual:processPath] ||
-        // Fix issue with iOS 8 `stringByStandardizingPath` removing leading
-        // `/private` path (when not running in the debugger or simulator only)
-        [imagePath hasPrefix:processPath]) {
+            // Fix issue with iOS 8 `stringByStandardizingPath` removing leading
```
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Utils/MSErrorLogFormatter.m]: This format doesn't look good.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/MSWrapperExceptionManagerInternal.h [hunk 2]
```diff
@@ -10,34 +10,50 @@
 @property NSMutableDictionary *wrapperExceptionData;
 @property NSData *unsavedWrapperExceptionData;
 @property CFUUIDRef currentUUIDRef;
-@property(weak, nonatomic) id<MSWrapperCrashesInitializationDelegate> crashesDelegate;
+@property(weak, nonatomic) id <MSWrapperCrashesInitializationDelegate> crashesDelegate;
```
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/MSWrapperExceptionManagerInternal.h]: We should have docs for properties and methods in the header. It was done by @achocron.

File: Global.xcconfig [hunk 3]
```diff
@@ -21,6 +26,34 @@ OTHER_CFLAGS[config=Debug][sdk=iphoneos*] = $(OTHER_CFLAGS) -fembed-bitcode-mark
...
 
+// :Mark: Extension API Only
+// Make sure we don't use API that are not available in extensions.
+// See https://pewpewthespells.com/blog/buildsettings.html for info about the flag.
 APPLICATION_EXTENSION_API_ONLY = YES;
 
+// :Mark: GCC_PREPROCESSOR_DEFINITIONS
 GCC_PREPROCESSOR_DEFINITIONS = $(inherited) $(XCODEBUILD_GCC_PREPROCESSOR_DEFINITIONS) MOBILE_CENTER_CONFIGURATION_$(CONFIGURATION) MOBILE_CENTER_NAME="@\""$(SDK_NAME)"\"" MOBILE_CENTER_VERSION="@\""$(VERSION_STRING)"\"" MOBILE_CENTER_BUILD="@\""$(BUILD_NUMBER)"\"" MOBILE_CENTER_C_NAME="\""$(SDK_NAME)"\"" MOBILE_CENTER_C_VERSION="\""$(VERSION_STRING)"\"" MOBILE_CENTER_C_BUILD="\""$(BUILD_NUMBER)"\""
+
+// :MARK: Warnings
```
Comment (jaelim-ms) [inline Global.xcconfig]: I remember @clpolet mentioned about this.

File: MobileCenter/MobileCenterTests/MSChannelDefaultTests.m [hunk 4]
```diff
@@ -1,7 +1,7 @@
-#import <Foundation/Foundation.h>
+@import Foundation;
 #import <OCHamcrestIOS/OCHamcrestIOS.h>
 #import <OCMock/OCMock.h>
```
Comment (jaelim-ms) [inline MobileCenter/MobileCenterTests/MSChannelDefaultTests.m]: `OCHamcrestIOS` and `OCMock` are not the case for this change?

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Utils/MSErrorLogFormatter.m [hunk 5]
```diff
@@ -709,15 +711,15 @@ + (MSBinaryImageType)imageTypeForImagePath:(NSString *)imagePath processPath:(NS
...
+                    // `/private` path (when not running in the debugger or simulator only)
+                    [imagePath hasPrefix:processPath]) {
       imageType = MSBinaryImageTypeAppBinary;
     } else if ([standardizedImagePath hasPrefix:appBundleContentsPath] ||
-               // Fix issue with iOS 8 `stringByStandardizingPath` removing
-               // leading `/private` path (when not running in the debugger or
-               // simulator only)
-               [imagePath hasPrefix:appBundleContentsPath]) {
+            // Fix issue with iOS 8 `stringByStandardizingPath` removing
```
Comment (clpolet) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Utils/MSErrorLogFormatter.m]: This one is weird too, I am wondering what's causing this. Maybe try to move the comments out of the condition.

File: MobileCenterCrashes/MobileCenterCrashesTests/MSAppleErrorLogTests.m [hunk 6]
```diff
@@ -1,6 +1,7 @@
-#import <Foundation/Foundation.h>
+@import Foundation;
+@import XCTest;
+
```
Comment (clpolet) [inline MobileCenterCrashes/MobileCenterCrashesTests/MSAppleErrorLogTests.m]: Which rules are you using to sort out the import?

File: Global.xcconfig (hunk 3 above)
Comment (TroubleMakerBen) [inline Global.xcconfig]: @jaelim-ms this is addressed in the other PR.

File: MobileCenterCrashes/MobileCenterCrashesTests/MSAppleErrorLogTests.m (hunk 6 above)
Comment (TroubleMakerBen) [inline MobileCenterCrashes/MobileCenterCrashesTests/MSAppleErrorLogTests.m]: 1. `#import` before `@module`
2. Sort both kinds alphabetically.

**Comment by TroubleMakerBen** [comment]
This branch is not based on `develop` but it is based on #269.

I'll merge that one to `/develop` first and then address your comments.

**Comment by TroubleMakerBen** [comment]
Closing this one. See #271

**Pull request closed without merging**